kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/metrics/MetricsCollector.java

```diff
@@ -0,0 +1,387 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.systemtests.metrics;
+
+import java.io.IOException;
+import java.security.InvalidParameterException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import io.fabric8.kubernetes.api.model.LabelSelector;
+
+import io.kroxylicious.systemtests.Constants;
+import io.kroxylicious.systemtests.executor.Exec;
+import io.kroxylicious.systemtests.resources.ComponentType;
+import io.kroxylicious.systemtests.utils.TestUtils;
+
+import static io.kroxylicious.systemtests.k8s.KubeClusterResource.cmdKubeClient;
+import static io.kroxylicious.systemtests.k8s.KubeClusterResource.kubeClient;
+
+/**
+ * The type Metrics collector.
+ */
+public class MetricsCollector {
+
+    private static final Logger LOGGER = LogManager.getLogger(MetricsCollector.class);
+
+    private final String namespaceName;
+    private final String scraperPodName;
+    private final ComponentType componentType;
+    private final String componentName;
+    private final int metricsPort;
+    private final String metricsPath;
+    private final LabelSelector componentLabelSelector;
+    private Map<String, String> collectedData;
+
+    /**
+     * The type Builder.
+     */
+    public static class Builder {
```

Comment:
does the build need to be public?

Reply:
The way it is used in the beforeEach in the MetricsST class is needed to be public. I didn't waste time to review all what is done in Strimzi, I have just copied and adapted as much as possible.